src/useChat.tsx

```diff
@@ -283,8 +351,13 @@ export const useChat = (initValues?: ChatMessage[], initModel?: string) => {
     });
   }, [fetchConversations]);
 
-  return useMemo(
-    () => ({
+  return useMemo(() => {
+    const providerInfo =
+      walletConnection.isWalletConnected && walletConnection.rdns
+        ? availableProviders.find((p) => p.info.rdns === walletConnection.rdns)
+            ?.info
+        : undefined;
```

Comment:
Check that the provider in the wallet store matches one of those announced by the window event